plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeBasic.java

```diff
@@ -2355,6 +2359,107 @@ public void testUnsupportedWriterVersion()
                 "Cannot execute vacuum procedure with 8 writer version");
     }
 
+    /**
+     * @see deltalake.clone_merge
+     */
+    @Test
+    public void testMergeOnClonedTable()
+            throws Exception
+    {
+        testMergeOnClonedTable("deltalake/clone_merge/clone_merge_source",  "deltalake/clone_merge/clone_merge_cloned", "clone_merge_source");
+        testMergeOnClonedTable("deltalake/clone_merge/clone_merge_deletion_vector_source",  "deltalake/clone_merge/clone_merge_deletion_vector_cloned", "clone_merge_deletion_vector_source");
+    }
+
+    private void testMergeOnClonedTable(String sourceResourceName, String clonedResourceName, String sourceTableDir)
+            throws Exception
+    {
+        String sourceTable = sourceTableDir + randomNameSuffix();
+        // load source table to a random suffix sub dir of the catalogDir
+        Path sourceLocation = catalogDir.resolve("clone_test_dir" + randomNameSuffix()).resolve(sourceTableDir);
+        FILE_SYSTEM.createDirectory(Location.of(sourceLocation.toString()));
+        copyDirectoryContents(new File(Resources.getResource(sourceResourceName).toURI()).toPath(), sourceLocation);
+        assertUpdate("CALL system.register_table(CURRENT_SCHEMA, '%s', '%s')".formatted(sourceTable, sourceLocation.toUri()));
+
+        @Language("SQL") String sourceTableValues = """
+                VALUES
+                (1, 'A', TIMESTAMP '2024-01-01'),
+                (2, 'B', TIMESTAMP '2024-01-01'),
+                (3, 'C', TIMESTAMP '2024-02-02'),
+                (4, 'D', TIMESTAMP '2024-02-02')
+                """;
+
+        assertQuery("SELECT * FROM " + sourceTable, sourceTableValues);
+
+        String clonedTable = "test_clone_merge_cloned_" + randomNameSuffix();
+        Path cloneTableLocation = sourceLocation.resolveSibling(clonedTable);
+        copyDirectoryContents(new File(Resources.getResource(clonedResourceName).toURI()).toPath(), cloneTableLocation);
+
+        // Replace the fixed s3 prefix with actual(local) absolute path prefix
+        updateClonedTableDeletionVectorPathPrefix(cloneTableLocation, sourceLocation.toString());
```

Comment:
You would need to update cloned table name in metadata, currently this test returns `Optional.empty()` from `shallowCloneSourceTableLocation`

Reply:
You are right, thanks for pointing out